contrib/DEVELOPMENT_WORKFLOW.md

```diff
@@ -1,159 +1,328 @@
 
-**master**
-- This is Bittensor's live production branch, which should only be updated by the core development team. This branch is protected, so refrain from pushing or merging into it unless authorized.
+### How the different components interact?
 
-**staging**
-- This branch is continuously updated and is where you propose and merge changes. It's essentially Bittensor's active development branch.
+Neurons in the Bittensor network interact with the Metagraph, a decentralized ledger that keeps track of the state of the network. The wallet is used to manage the tokens that are used for incentives in the network. The subtensor is a lower-level protocol that handles communication between neurons.
 
-## Development Model
+# 2. Main Components
+### Bittensor Protocol
 
-### Feature Branches
+The Bittensor protocol is the backbone of the network. It defines how neurons communicate with each other and with the Metagraph.
 
-- Branch off from: `staging`
-- Merge back into: `staging`
-- Naming convention: `feature/<ticket>/<descriptive-sentence>`
+### Neurons
 
-Feature branches are used to develop new features for upcoming or future releases. They exist as long as the feature is in development, but will eventually be merged into `staging` or discarded. Always delete your feature branch after merging to avoid unnecessary clutter.
+Neurons are the nodes in the Bittensor network. They represent AI models that are learning from the network.
 
-### Release Branches
+### Subtensor
 
-- Branch off from: `staging`
-- Merge back into: `staging` and then `master`
-- Naming convention: `release/<version>/<descriptive-message>/<creator's-name>`
+Subtensor is a lower-level protocol that handles communication between neurons. It ensures that data is transfered securely and efficiently across the network.
 
-Release branches support the preparation of a new production release, allowing for minor bug fixes and preparation of metadata (version number, configuration, etc). All new features should be merged into `staging` and wait for the next big release.
+### Metagraph
 
-### Hotfix Branches
+The Metagraph is a decentralized ledger that keeps track of the state of the Bittensor network. It records which neurons are part of the network and how they are connected.
 
-General workflow:
 
-- Branch off from: `master` or `staging`
-- Merge back into: `staging` then `master`
-- Naming convention: `hotfix/<version>/<descriptive-message>/<creator's-name>` 
+# 3. Developer Notes
 
-Hotfix branches are meant for quick fixes in the production environment. When a critical bug in a production version must be resolved immediately, a hotfix branch is created.
 
-## Git Operations
+Notes for Python Script Style
+------------------
 
-#### Create a feature branch
+### 1. General Coding Style
```

Comment:
This is also better suited to style guide. Potentially integrate (not just copy paste) there?